samples/enums/is_variant_binding.jakt

```diff
@@ -10,8 +10,8 @@ function main() {
     let foo = Foo::Bar(5)
     let baz = Foo::Baz(m: "Hello")
 
-    if foo is Foo::Bar(n) {
-        if baz is Foo::Baz(m: different) {
+    if foo is Bar(n) {
```

Comment:
Just tested here. Right now, it works just fine (it returns a bool).
I am thinking that maybe we could create destructure assignment for enum variants?

Something like this:
```
let (n) = foo is Bar(n) // if the type of n is i64, it could the binding could be of type 164?
```

Don't know if is too crazy of an idea.

Reply:
and what happens to the binding?
since there's no particularly meaningful thing that can be done with that expression, we should disallow it.

re destructuring enums, I don't see how that would help the ergonomics, you'll end up with an if/match anyway:
```
let Bar(n) = foo
if n.has_value() { ... }
// or
match n { ... }
```
vs just
```
if foo is Bar(n) { ... }
// or
match foo { Bar(n) => { ... } }
```

Reply:
@adleroliveira I think `let (n) = foo // if the type of n is i64, the binding could be of type 164?` is maybe a step too far, as it hurts readability.

As long as we have the containing type or enum variant in the code, should be quite a bit easier to read.